lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  DescribeAddonVersionsCommand,
+  DescribeAddonVersionsCommandInput,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { cache } from '../../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { EKSAddonsFilter } from './types';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly caching = true;
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const filter: EKSAddonsFilter = JSON.parse(serializedFilter);
+    const eksClient = this.getEKSClient(filter);
```

Comment:
why do we need a new client on each request? this can cause a massive slowdown

Reply:
AWS EKS is a regional service. In practice, we can expect that EKS should make Addon versions consistent across all commercial regions, but in theory they could choose to release a version in selected regions only. So I decided to allow users to optionally specify AWS `region` per EKS addon "package", which would require rebuilding the EKS client per request.

The code is similar to AWS AMI datasource: https://github.com/jzhn/renovate/blob/feat/23410_aws-eks-addon-support/lib/modules/datasource/aws-machine-image/index.ts#L93. 

If we build with the assumption that EKS addon versions are consistent, we could have a singleton EKSClient. but I feel this might cause confusion for users.

Reply:
then cache the client per region, so it's reusable.

Reply:
Updated to cache clieint per region + profile.